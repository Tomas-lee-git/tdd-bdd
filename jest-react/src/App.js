import React from 'react';
import './App.css';

function App() {
  return <div className="App" data-test="app">hello,world</div>;
}

export default App;
