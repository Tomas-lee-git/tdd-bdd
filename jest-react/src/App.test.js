import React from 'react';
// import { render } from '@testing-library/react';
import Enzyme, { shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from './App';

Enzyme.configure({ adapter: new Adapter() });

test('renders hello world', () => {
  // const { getByText } = render(<App />);
  // const content = getByText(/hello,world/i);
  // expect(content).toBeInTheDocument();

  // shallow 只渲染<App/>组件本身，不关心<App/>的子组件
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toExist();
});
