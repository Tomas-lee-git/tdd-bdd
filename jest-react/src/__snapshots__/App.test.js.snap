// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders hello world 1`] = `
<div
  className="App"
  data-test="app"
>
  hello,world
</div>
`;
